### DISCUSSION TITLE
Validate a schema file: Find typos in property names

### DISCUSSION THREAD

**Discussion by cweiske** [json-schema-org/community#380] [Q&A] [answered]
I am writing a json schema file and want to validate that is is correct.
For that I downloaded the schema itself from `https://json-schema.org/draft/2020-12/schema` as well as all meta dependency files.
Then I installed check-jsonschema via pip and try to validate it:

```
$ check-jsonschema --schemafile ~/dev/tools/json-schema-schema/json-schema.schema.json gamestick-game.schema.json 
ok -- validation done
```

Now my problem: It always tells me that my schema is valid.
I renamed the top-level "title" to "titles", "required" to "requireds" and the schema is always valid.

How can I be sure that I do not have any typos in my property names, when they are always considered valid?

**Comment by gregsdennis** [comment]
The current meta-schemas (the schema that validates schemas) don't check for mistyped keywords.  They instead allow unknown keywords without checking them.  So when you mistype a keyword, it's as if that keyword just isn't there.

**Reply by cweiske** [reply]
That's .. not very helpful :(

**Reply by Relequestual** [reply]
It has been pointed out previously, and we are discussing this exact thing currently.

**Comment by jdesrosiers** [comment, marked as answer]
In JSON Schema 2020-12 (and all versions that came before it), it's allowed for users to add their own custom keywords. The meta-schema isn't capable of being smart enough to know that `requireds` is a typo and not a custom keyword that you chose to include in your schema.

IMO, what's really needed for something like this is a linter that _is_ smart enough to notice that `requireds` looks a whole lot like the known keyword `required` and give you a warning that it's probably a typo. Unfortunately, a linter capable of checking for those sorts of errors doesn't exist. There's been talk about it for years, but no one has managed to find the time to build it.

That said, there's an easy workaround you can use to validate that your schemas don't have any custom/misspelled keywords. You can use a custom meta-schema.

```json
{
  "$id": "https://example.com/draft/2020-12/schema/strict",
  "$schema": "https://json-schema.org/draft/2020-12/schema",

  "$vocabulary": {
    "https://json-schema.org/draft/2020-12/vocab/core": true,
    "https://json-schema.org/draft/2020-12/vocab/applicator": true,
    "https://json-schema.org/draft/2020-12/vocab/unevaluated": true,
    "https://json-schema.org/draft/2020-12/vocab/validation": true,
    "https://json-schema.org/draft/2020-12/vocab/meta-data": true,
    "https://json-schema.org/draft/2020-12/vocab/format-annotation": true,
    "https://json-schema.org/draft/2020-12/vocab/content": true
  },
  "$dynamicAnchor": "meta",

  "$ref": "https://json-schema.org/draft/2020-12/schema",
  "unevaluatedProperties": false
}
```

You can validate your schemas against this one or use it's `$id` as the `$schema` and most implementations will validate your schema against this schema automatically.

```json
{
  "$id": "https://example.com/gamestick-game.schema.json",
  "$schema": "https://example.com/draft/2020-12/schema/strict",

  ...
}
```

**Reply by cweiske** [reply]
Thank you! That works fine.
